model/proto/metrics/service.proto

```diff
@@ -0,0 +1,131 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax="proto3";
+
+package jaeger.api_v2;
+
+import "otelmetric.proto";
+import "otelspankind.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+option go_package = "api_v2";
+option java_package = "io.jaegertracing.api_v2";
+
+// Enable gogoprotobuf extensions (https://github.com/gogo/protobuf/blob/master/extensions.md).
+// Enable custom Marshal method.
+option (gogoproto.marshaler_all) = true;
+// Enable custom Unmarshal method.
+option (gogoproto.unmarshaler_all) = true;
+// Enable custom Size method (Required by Marshal and Unmarshal).
+option (gogoproto.sizer_all) = true;
+
+// MetricsQueryBaseRequest is the base request parameter accompanying a MetricsQueryService RPC call.
+message MetricsQueryBaseRequest {
+  // service_names are the service names to fetch metrics from.
+  // The results will be grouped by service_name.
+  // Required. At least one service name must be provided.
+  repeated string service_names = 1;
+
+  // groupByOperation determines if the metrics returned should be grouped by operation.
+  // Optional. Default = false.
+  bool groupByOperation = 2;
+
+  // end_time is the ending time of the time series query range.
+  // Optional. Default = now.
+  google.protobuf.Timestamp end_time = 3 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = true
+  ];
+
+  // lookback is the duration from the end_time to look back on for metrics data points.
+  // For example, if set to 1h, the query would span from end_time-1h to end_time.
+  // Optional. Default = 1h.
+  google.protobuf.Duration lookback = 4 [
+    (gogoproto.stdduration) = true,
+    (gogoproto.nullable) = true
+  ];
+
+  // step size is the duration between data points of the query results.
+  // For example, if set to 5s, the results would produce a data point every 5 seconds
+  // from the start_time to end_time.
+  // Optional. Default = 5s.
+  google.protobuf.Duration step = 5 [
+    (gogoproto.stdduration) = true,
+    (gogoproto.nullable) = true
+  ];
+
+  // ratePer is the duration in which the per-second rate of change is calculated for a cumulative counter metric.
+  // Optional. Default = 10m.
+  google.protobuf.Duration ratePer = 6 [
+    (gogoproto.stdduration) = true,
+    (gogoproto.nullable) = true
+  ];
+
+  // spanKinds is the list of span kinds to include (logical OR) in the resulting metrics aggregation.
+  // Optional. Default = [SPAN_KIND_SERVER].
+  repeated SpanKind spanKinds = 7;
+}
+
+// GetLatenciesRequest contains parameters for the GetLatencies RPC call.
+message GetLatenciesRequest {
+  MetricsQueryBaseRequest baseRequest = 1;
+  // quantile is the quantile to compute from latency histogram metrics.
```

Comment:
Could use an example. What are the units? If we want p99, should this be `quantile=99`? If so, why the double type?

Reply:
> Could use an example. What are the units?

+1 I'll add an example.

> If we want p99, should this be quantile=99? If so, why the double type?

The possible values range from 0-1; so p99 would be `0.99`. Note that p99 is a percentile, which ranges from 0-100 and is a 100-quantile. 

IIUC quantiles are a broader definition which equally slices the population of data up into any number. For example, p999 = 99.9th percentile and would have a quantile value of 0.999. 

Moreover, most (if not all) metrics backends use quantiles instead of percentiles as parameters to their "histogram quantile" calculations, for instance:
- [InfluxDB](https://docs.influxdata.com/influxdb/cloud/reference/flux/stdlib/built-in/transformations/aggregates/histogramquantile/)
- [Prometheus](https://prometheus.io/docs/practices/histograms/#quantiles), [VictoriaMetrics](https://docs.victoriametrics.com/MetricsQL.html) and M3

Maintaining the 0-1 range of values for quantile means there is no need to perform any computation to map between percentiles to quantiles before passing it the metrics backend, as well as supporting quantiles more than 100, which would be a rare use case, but I feel there's no harm in doing so, as long as it's well documented.